File: backend/src/main/java/com/ruschlang/backend/domain/restaurant/entity/Restaurant.java
package com.ruschlang.backend.domain.restaurant.entity;

import com.ruschlang.backend.domain.review.entity.Review;
import com.ruschlang.backend.global.common.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "restaurants")
public class Restaurant extends BaseTimeEntity {

    @Id
    @Column(columnDefinition = "char(36)")
    private String id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(nullable = false, length = 255)
    private String address;

    @Column(nullable = false, precision = 10, scale = 6)
    private BigDecimal lat;

    @Column(nullable = false, precision = 10, scale = 6)
    private BigDecimal lng;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Lob
    @Column(name = "photo_url", columnDefinition = "LONGTEXT")
    private String photoUrl;

    @OneToMany(mappedBy = "restaurant", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Review> reviews = new ArrayList<>();

    @PrePersist
    public void prePersist() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
    }

    @Builder
    public Restaurant(String name, String address, BigDecimal lat, BigDecimal lng, String description, String photoUrl) {
        this.name = name;
        this.address = address;
        this.lat = lat;
        this.lng = lng;
        this.description = description;
        this.photoUrl = photoUrl;
    }
}

File: frontend/src/types/restaurant.ts
export interface Restaurant {
  id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  description: string;
  photoUrl: string;
  createdAt: string;
  updatedAt: string;
  reviews: Review[];
}

export interface Review {
  id: string;
  restaurantId: string;
  name: string;
  generation?: number;
  rating: number;
  note: string;
  photoUrl: string;
  createdAt: string;
}

export interface RestaurantCreateRequest {
  name: string;
  address: string;
  lat: number;
  lng: number;
  description: string;
  photoUrl: string;
}

export interface ReviewCreateRequest {
  name: string;
  generation?: number;
  rating: number;
  note: string;
  photoUrl: string;
}
